### PR TITLE
ROC-2476: calling api to reject or accept an offer

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-2476

### Change description ###
Calling API endpoints to store informations about 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/claims/claimStoreClient.ts [hunk 1]
```diff
@@ -45,6 +47,27 @@ export class ClaimStoreClient {
       }
     })
   }
+
+  static acceptOfferByUser (acceptedBy: string, user: User): Promise<Claim> {
```
Comment (rlewan) [inline src/main/app/claims/claimStoreClient.ts]: Would be nice to rename methods in this class to have just `makeOffer`, acceptOffer` and `rejectOffer`.

Comment (rlewan) [reply src/main/app/claims/claimStoreClient.ts]: What do you think about having a separate client for it as well?

File: src/test/http-mocks/claim-store.ts [hunk 2]
```diff
@@ -236,11 +241,25 @@ export function rejectSaveOfferForDefendant (reason: string = 'HTTP error') {
...
 }
+
 export function resolveSaveOffer () {
   mock(`${serviceBaseURL}/claims`)
     .post(new RegExp('/[0-9]+/offers/defendant'))
-    .reply(HttpStatus.OK)
+    .reply(HttpStatus.CREATED)
 }
+
+export function resolveAcceptOffer (by: string = 'claimant') {
```
Comment (rlewan) [inline src/test/http-mocks/claim-store.ts]: What do you think about testing for failures as well?

Comment (geek-soft) [inline src/test/http-mocks/claim-store.ts]: Should't that be in capitals? Would be possible to reuse `MadeBy` buy repurposing the class so we have `OfferParticipantType.CLAIMANT.value`. I am pretty sure we could find better name as well :)

File: src/main/app/claims/claimStoreClient.ts (hunk 1 above)
Comment (yarpo) [reply src/main/app/claims/claimStoreClient.ts]: I was thinking about it as we have a separate client for CCJ as well.

File: src/test/http-mocks/claim-store.ts (hunk 2 above)
Comment (yarpo) [reply src/test/http-mocks/claim-store.ts]: It doesn't matter if it's upper or lower case. Both works for the same endpoint.

This is actually weird we don't have consts like this so far on frontend. If we want to introduce it what do you think about MadeBy enum (as we have one in API)? The name is more generic and we can use it in more contexts.

**Pull request merged**